torchx/specs/api.py

```diff
@@ -702,6 +702,7 @@ class runopt:
     opt_type: Type[CfgVal]
     is_required: bool
     help: str
+    creator: Optional[Callable[[CfgVal], CfgVal]] = None
```

Comment:
Its not clear how to use this `creator`. Can/should I use a `creator` for a primitive `CfgVal` (eg ones that are not enums)?

In general schedule options were kept dead simple because they are meant to be directly parsable from command-line (after all torchx is mostly used from CLI). This is why they are not structured, and only basic type checking is done for sanity. Enums are tricky to support especially when the values are sent over the wire and as such support for enums in scheduler opts should be well motivated and thorougly justified as we would be trading off simplicity/clarity for <whatever the motivation is>.

I'd like to understand the motivation behind wanting to support enums for scheduler opts before we add them. cc @d4l3k